src/main/java/org/opensearch/sdk/GetNetty4Transport.java

```diff
@@ -0,0 +1,122 @@
+import org.opensearch.transport.ConnectionManager;
+import org.opensearch.transport.SharedGroupFactory;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.netty4.Netty4Transport;
+
+import static java.util.Collections.emptySet;
+import static org.opensearch.common.UUIDs.randomBase64UUID;
+
+/**
+ * This class initializes a Netty4Transport object and control communication between the extension and OpenSearch.
+ */
+
+public class GetNetty4Transport {
+
+    /**
+     * Initializes a Netty4Transport object. This object will be wrapped in a {@link TransportService} object.
+     *
+     * @param settings  The transport settings to configure.
+     * @param threadPool  A thread pool to use.
+     * @return The configured Netty4Transport object.
+     */
+    public Netty4Transport getNetty4Transport(Settings settings, ThreadPool threadPool) {
+        NetworkService networkService = new NetworkService(Collections.emptyList());
+        PageCacheRecycler pageCacheRecycler = new PageCacheRecycler(settings);
+        IndicesModule indicesModule = new IndicesModule(Collections.emptyList());
+        SearchModule searchModule = new SearchModule(settings, Collections.emptyList());
+
+        List<NamedWriteableRegistry.Entry> namedWriteables = Stream.of(
+            NetworkModule.getNamedWriteables().stream(),
+            indicesModule.getNamedWriteables().stream(),
+            searchModule.getNamedWriteables().stream(),
+            null,
+            ClusterModule.getNamedWriteables().stream()
+        ).flatMap(Function.identity()).collect(Collectors.toList());
+
+        final NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry(namedWriteables);
+
+        final CircuitBreakerService circuitBreakerService = new NoneCircuitBreakerService();
+
+        Netty4Transport transport = new Netty4Transport(
+            settings,
+            Version.CURRENT,
+            threadPool,
+            networkService,
+            pageCacheRecycler,
+            namedWriteableRegistry,
+            circuitBreakerService,
+            new SharedGroupFactory(settings)
+        );
+
+        return transport;
+    }
+
+    /**
+     * Initializes the TransportService object for this extension. This object will control communication between the extension and OpenSearch.
+     *
+     * @param settings  The transport settings to configure.
+     * @param extensionsRunner method to call
+     * @return The initialized TransportService object.
+     */
+    public TransportService initializeExtensionTransportService(Settings settings, ExtensionsRunner extensionsRunner) {
+
+        ThreadPool threadPool = new ThreadPool(settings);
+
+        Netty4Transport transport = getNetty4Transport(settings, threadPool);
+
+        final ConnectionManager connectionManager = new ClusterConnectionManager(settings, transport);
+
+        // Stop any existing transport service
```

Comment:
I added this code block while it was in the ExtensionsRunner class.  Now that it's in its own class we have much better control over its initialization so you can remove these lines.